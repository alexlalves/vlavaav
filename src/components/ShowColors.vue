<template>
  <div>
    <div class="show-colors">
      <div
        v-for="color in colors"
        :key="color"
        :style="`background-color: #${color};`"
        @click="copyToClipboard(color)"
        class="show-colors__cell"
      >
        <div class="show-colors__cell-description-box">
          <code>{{ color }}</code>
        </div>
      </div>

      <div
        @click="copyLinkToClipboard()"
        class="show-colors__cell"
      >
        <div class="show-colors__cell-description-box">
          <code>Share this palette</code>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class ShowColors extends Vue {
  @Prop(
    { required: false, type: String, default: () => '' },
  ) readonly commaSeparatedColors!: string;

  get colors() {
    return this.commaSeparatedColors.split(',');
  }

  public copyToClipboard(text: string) {
    navigator.clipboard.writeText(text);
  }

  public copyLinkToClipboard() {
    this.copyToClipboard(document.URL);
  }
}
</script>

<style scoped lang="scss">
.show-colors {
  display: grid;
  grid-gap: 16px;
  grid-template-columns: 1fr 1fr;

  @media (min-width: 480px) {
    grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
  }
}

.show-colors__cell {
  align-items: flex-end;
  cursor: pointer;
  display: flex;
  height: 160px;

  &:hover {
    transform: scale(1.05);
  }
}

.show-colors__cell-description-box {
  background-color: rgba(12, 12, 12, 0.8);
  font-size: 1.25em;
  padding: .5em;
  width: 100%;
}
</style>
