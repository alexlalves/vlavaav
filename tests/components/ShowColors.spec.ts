import { shallowMount } from '@vue/test-utils';
import ShowColors from '@/components/ShowColors.vue';

describe('InitialInfo', () => {
  def('colors', () => 'FFF,FF0');

  const wrapper = () => shallowMount(
    ShowColors, {
      propsData: {
        commaSeparatedColors: get('colors'),
      },
      stubs: ['router-link'],
    },
  );

  it('matches snapshot', () => {
    expect(wrapper().element).toMatchSnapshot();
  });

  describe('numberOfColumns', () => {
    describe('when the number of colors is almost a perfect square', () => {
      def('colors', () => [
        'white', 'yellow',
        'blue',
      ].join(','));

      it('leaves space for the copy link', () => {
        expect(wrapper().vm.numberOfColumns).toEqual(2);
      });
    });

    describe('when the number of colors is bigger than a perfect square - 1', () => {
      def('colors', () => [
        'white', 'yellow',
        'blue', 'green',
      ].join(','));

      it('retuns an extra column', () => {
        expect(wrapper().vm.numberOfColumns).toEqual(3);
      });
    });
  });
});
