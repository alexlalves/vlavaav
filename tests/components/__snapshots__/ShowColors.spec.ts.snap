// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialInfo matches snapshot 1`] = `
<div>
  <div
    class="show-colors"
    style="grid-template-columns: repeat(2, 1fr);"
  >
    <div
      class="show-colors__cell"
      style="background-color: rgb(255, 255, 255); flex-grow: 1;"
    >
      <div
        class="show-colors__cell-description-box"
      >
        <code
          class="show-colors__cell-description"
        >
          
          FFF
          
          <br />
        </code>
      </div>
    </div>
    <div
      class="show-colors__cell"
      style="background-color: rgb(255, 255, 0); flex-grow: 1;"
    >
      <div
        class="show-colors__cell-description-box"
      >
        <code
          class="show-colors__cell-description"
        >
          
          FF0
          
          <br />
        </code>
      </div>
    </div>
     
    <div
      class="show-colors__cell"
    >
      <div
        class="show-colors__cell-description-box"
      >
        <code
          class="show-colors__cell-description"
        >
          
          Share this palette
          
          <br />
        </code>
      </div>
    </div>
  </div>
   
  <div
    class="show-colors__footer"
  >
    <router-link-stub
      class="show-colors__footer-link"
      to="[object Object]"
    >
      Click here
    </router-link-stub>
    
    to paste a new set of colors.
    
    <strong>
      VLAVAAV
    </strong>
     Â© Alex L. Alves 2021.
  
  </div>
</div>
`;
